### PR TITLE
Test reproducibility for any seed

### PR DESCRIPTION
**Pull request by toslunar**
- Test reproducibility for non-zero seeds
- Make `generate`, `generate_many` return different values on `repeat` and `retry`.

### CODE REVIEW

**Comment by niboshi** [comment]
LGTM
jenkins, test this please

File: tests/cupy_tests/random_tests/test_generator.py [hunk 1]
```diff
@@ -19,15 +19,16 @@ class RandomGeneratorTestCase(unittest.TestCase):
     target_method = None
 
     def setUp(self):
-        self.rs = generator.RandomState(seed=testing.generate_seed())
+        self.__seed = testing.generate_seed()
```
Comment (okuta) [inline tests/cupy_tests/random_tests/test_generator.py]: Why do you use double underscore?

Comment (toslunar) [reply tests/cupy_tests/random_tests/test_generator.py]: Because a child class does not need to know the seed to reset the random state.

Comment (okuta) [reply tests/cupy_tests/random_tests/test_generator.py]: OK

**Comment by okuta** [comment]
LGTM!

**Pull request merged**